js/pin.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+(function () {
+  var mapPins = document.querySelector('.map__pins');
+
+  var renderPin = function (pin) {
+    var pinTemplate = document.querySelector('#pin').content.querySelector('.map__pin');
+    var pinElement = pinTemplate.cloneNode(true);
+
+    pinElement.style = 'left:' + (pin.location.x - (window.constant.HALF_PIN_WIDTH)) + 'px; top:' + (pin.location.y - window.constant.PIN_HEIGHT) + 'px;';
+    pinElement.querySelector('img').src = pin.author.avatar;
+    pinElement.querySelector('img').alt = pin.offer.title;
+
+    var onMapPinsClick = function () {
+      var mapCard = document.querySelector('.map__card');
+      if (mapCard) {
+        mapCard.remove();
+      }
+
+      window.card.makeCardElements(pin);
+      mapCard = document.querySelector('.map__card');
+      var mapCardClose = mapCard.querySelector('.popup__close');
+
+      var closePopup = function () {
+        mapCard.remove();
+        document.removeEventListener('keydown', onMapCardClosePress);
+      };
+
+      var onMapCardCloseClick = function () {
+        closePopup();
+      };
+
+      var onMapCardClosePress = function (evt) {
+        if (evt.key === window.constant.ESC_KEY) {
+          closePopup();
+        }
+      };
+
+      mapCardClose.addEventListener('click', onMapCardCloseClick);
+      document.addEventListener('keydown', onMapCardClosePress);
+    };
+
+    pinElement.addEventListener('click', onMapPinsClick);
+
+    return pinElement;
+  };
+
+  var makePinElements = function (pins) {
+    var fragment = document.createDocumentFragment();
+    pins.forEach(function (element) {
+      fragment.appendChild(renderPin(element));
+    });
+    mapPins.appendChild(fragment);
+  };
+
+  window.pin = {
+    makePinElements: makePinElements
+  };
```

Comment:
В имени свойства не должно содержаться имя объекта, можно переименовать так
```suggestion
  window.pin = {
    makeElements: makePinElements
  };
```